samples/demo-asset-index/src/main/java/org/eclipse/dataspaceconnector/demo/assets/DemoAssetExtension.java

```diff
@@ -0,0 +1,58 @@
+/*
+ *  Copyright (c) 2021 Daimler TSS GmbH
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Daimler TSS GmbH - Initial API and Implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.demo.assets;
+
+import org.eclipse.dataspaceconnector.spi.asset.AssetIndex;
+import org.eclipse.dataspaceconnector.spi.monitor.Monitor;
+import org.eclipse.dataspaceconnector.spi.system.ServiceExtension;
+import org.eclipse.dataspaceconnector.spi.system.ServiceExtensionContext;
+
+import java.util.Set;
+
+public class DemoAssetExtension implements ServiceExtension {
+    private static final String NAME = "Demo Asset Index Extension";
+    private static final String[] PROVIDES = {
+            AssetIndex.class.getName()
```

Comment:
also see issue #79. nothing to do at the moment, just something to keep in mind.